docs/architecture/ADR-003-Non-interactive-defaults.md

```diff
@@ -0,0 +1,49 @@
+# ADR 003: Non-interactive Defaults, Wrapped Transactions, and Subtree Root Message Inclusion Checks
+
+## Intro
+
+Currently, when checking for message inclusion, validators recreate the share commitment from the messages in the block and compare those with what are signed over in the `MsgPayForData` transactions also in that block. If any commitment is not found in one of the PFD transactions, or if there is a commitment that doesn't have a corresponding message, then they reject that block.
+
+While this functions as a message inclusion check, the light client has to assume that 2/3's of the voting power is honest in order to be assured that both the messages they are interested in and the rest of the messages paid for in that block are actually included.
+
+To fix this, the spec outlines the “non-interactive default rules”. These involve a few additional message layout rules that allow for commitments to messages to consist entirely of sub tree roots of the data root, and for those sub tree roots to be generated only from the message itself (so that the user can sign something “non-interactively”). NOTE: MODIFIED FROM THE SPEC
+
+
+- Messages begin at a location aligned with the largest power of 2 that is not larger than the message length or k.
+- If the largest power of 2 of a given message spans multiple rows it must begin at the start of a row (this can occur if a message is longer than k shares or if the block producer decides to start a message partway through a row and it cannot fit).
```

Comment:
[question] I'm confused by this rule's application in `example-full-block.png` for `ns=2`. `ns=2` is a message that spans 7 shares. The largest power of 2 less than the message length (7) or k (8) is 4. The message is aligned with index 4.

> If the largest power of 2 of a given message spans multiple rows

The largest power of 2 for this message (4) doesn't span multiple rows but this message does because the block producer decided to start this message partway through the row and it cannot fit. If `ns=2` is indeed supposed to be split across rows, then should this read:
```suggestion
- If the largest power of 2 of a given message is larger than k then it must begin at the start of a row.
```

If `ns=2` is supposed to start at a new row, I think the current language makes sense.

## example-full-block
![example-full-block](https://user-images.githubusercontent.com/3699047/187477722-e1a63e22-d397-4703-bd86-d0570053f8d8.png)

Reply:
Still thinking about this, but I think the first rule might be the only one we need. The we can just go into more detail of a what an "aligned power of two" actually means